### PR TITLE
Enable ruff's unused variable checking everywhere in pytorch

### PR DESCRIPTION
**Pull request by rec**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136965

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Comment by rec** [comment]
One of these errors is bogus; the other one, `Lint for torch/_functorch/_aot_autograd/collect_metadata_analysis.py`, is fixed in a parent commit but I don't want spam the job queue by re-requesting absolutely everything.

**Review by cyyever — APPROVED** [review]

**Comment by rec** [comment]
@Skylion007 - this is the last one!

Up for the honors?

File: torch/_inductor/codegen/cuda/cuda_template.py [hunk 1]
```diff
@@ -90,9 +90,7 @@ def generate(  # type: ignore[override]
             call_args,
             expected_args,
         )
-        extra_args = V.graph.sizevars.size_hints(
-            map(sympy.expand, call_args[len(expected_args) :])
-        )
+        V.graph.sizevars.size_hints(map(sympy.expand, call_args[len(expected_args) :]))
```
Comment (albanD) [inline torch/_inductor/codegen/cuda/cuda_template.py:93]: cc @eellison is this a bug and the variable should be used?

**Review by albanD — APPROVED** [review]
:tada: :tada: :tada: :tada: :tada:

**Pull request closed without merging**